sparkle/src/components/Button.tsx

```diff
@@ -1,261 +1,187 @@
-    },
-  },
-  secondary: {
-    base: "s-text-action-500 s-border-structure-200 s-bg-structure-0",
-    hover: "hover:s-bg-action-50 hover:s-border-action-200",
-    active: "active:s-bg-action-100 active:s-border-action-500",
-    disabled: "s-text-action-300 s-border-structure-200 s-bg-structure-0",
-    dark: {
-      base: "dark:s-text-action-600-dark dark:s-border-structure-200-dark dark:s-bg-structure-100-dark",
-      hover:
-        "dark:hover:s-text-action-700-dark dark:hover:s-bg-action-50-dark dark:hover:s-border-action-300-dark",
-      active:
-        "dark:active:s-bg-action-100-dark dark:active:s-border-action-500-dark",
-      disabled:
-        "dark:s-text-element-600-dark dark:s-border-structure-100-dark dark:s-bg-structure-200-dark",
-    },
-  },
-  secondaryWarning: {
-    base: "s-text-warning-500 s-border-structure-200 s-bg-structure-0",
-    hover: "hover:s-bg-warning-50 hover:s-border-warning-200",
-    active: "active:s-bg-warning-100 active:s-border-warning-500",
-    disabled: "s-text-warning-300 s-border-structure-200 s-bg-structure-0",
-    dark: {
-      base: "dark:s-text-warning-600-dark dark:s-border-structure-300-dark dark:s-bg-structure-100-dark",
-      hover:
-        "dark:hover:s-bg-warning-50-dark dark:hover:s-border-warning-300-dark",
-      active:
-        "dark:active:s-bg-warning-100-dark dark:active:s-border-warning-500-dark",
-      disabled:
-        "dark:s-text-element-600-dark dark:s-border-structure-100-dark dark:s-bg-structure-200-dark",
-    },
-  },
-  tertiary: {
-    base: "s-text-element-800 s-border-structure-200 s-bg-structure-0",
-    hover:
-      "hover:s-bg-action-50 hover:s-text-action-500 hover:s-border-action-200",
-    active:
-      "active:s-bg-action-100 active:s-text-action-600 active:s-border-action-500",
-    disabled: "s-text-element-500 s-border-structure-200 s-bg-structure-0",
-    dark: {
-      base: "dark:s-text-element-800-dark dark:s-border-structure-300-dark dark:s-bg-structure-100-dark",
-      hover:
-        "dark:hover:s-text-action-300 dark:hover:s-bg-action-50-dark dark:hover:s-border-action-300-dark",
-      active:
-        "dark:active:s-bg-action-100-dark dark:active:s-border-action-500-dark",
-      disabled:
-        "dark:s-text-element-600-dark dark:s-border-structure-100-dark dark:s-bg-structure-200-dark",
-    },
-  },
+const spinnerVariantsMap: Record<ButtonVariantType, SpinnerVariant> = {
+  primary: "light",
+  highlight: "light",
+  warning: "light",
+  outline: "dark",
+  ghost: "dark",
 };
 
-const transitionClasses =
-  "s-transition-all s-ease-out s-duration-200 s-cursor-pointer";
+const spinnerVariantsMapIsLoading: Record<ButtonVariantType, SpinnerVariant> = {
+  primary: "light",
+  highlight: "light",
+  warning: "light",
+  outline: "slate400",
+  ghost: "slate400",
+};
 
-const magnifyingClasses =
-  "hover:s-scale-105 hover:s-drop-shadow-md active:s-scale-100 active:s-drop-shadow-none";
+export interface MetaButtonProps
```

Comment:
We probably don't want to export this interface.

Reply:
Indeed, mistake